src/lib/Libdb/pgsql/pbs_db_utility

```diff
@@ -0,0 +1,1302 @@
+		ret=$?
+		if [ $ret -ne 0 ]; then
+			ret=$?
+			if [ $ret -ne 0 ]; then
+				return $ret
+			fi
+			return 1
+		fi
+		ver="1.1.0"
+	fi
+
+	${PBS_EXEC}/sbin/pbs_dataservice status > /dev/null
+	if [ $? -eq 1 ]; then
+		# not running, start now
+		${PBS_EXEC}/sbin/pbs_dataservice start > ${outfile}
+		if [ $? -ne 0 ]; then
+			cat ${outfile}
+			rm -f ${outfile}
+			ret=$?
+			if [ $ret -ne 0 ]; then
+				return $ret
+			fi
+			return 1
+		fi
+		rm -f ${outfile}
+	fi
+
+	if [ "$ver" = "1.1.0" ]; then
+		upgrade_pbs_schema_from_v1_1_0
+		ret=$?
+		if [ $ret -ne 0 ]; then
+			return $ret
+		fi
+		ver="1.2.0"
+	fi
+
+	if [ "$ver" = "1.2.0" ]; then
+		upgrade_pbs_schema_from_v1_2_0
+		ret=$?
+		if [ $ret -ne 0 ]; then
+			return $ret
+		fi
+		ver="1.3.0"
+	fi
+
+	if [ "$ver" = "1.3.0" ]; then
+		upgrade_pbs_schema_from_v1_3_0
+		ret=$?
+		if [ $ret -ne 0 ]; then
+			return $ret
+		fi
+		ver="1.4.0"
+	fi
+
+	if [ "$ver" = "1.4.0" ]; then
+		upgrade_pbs_schema_from_v1_4_0
+		ret=$?
+		if [ $ret -ne 0 ]; then
+			exit $ret
+		fi
+		ver="1.5.0"
+	else
+		echo "Cannot upgrade PBS datastore version $ver"
+		ret=$?
+		if [ $ret -ne 0 ]; then
+			return $ret
+		fi
+		return 1
+	fi
+
+	${PBS_EXEC}/sbin/pbs_dataservice status > /dev/null
+	if [ $? -eq 1 ]; then
+		# not running, start now
+		${PBS_EXEC}/sbin/pbs_dataservice start > ${outfile}
+		if [ $? -ne 0 ]; then
+			cat ${outfile}
+			rm -f ${outfile}
+			ret=$?
+			if [ $ret -ne 0 ]; then
+				return $ret
+			fi
+			return 1
+		fi
+	fi
+	rm -f ${outfile}
+
+	${PBS_EXEC}/sbin/pbs_dataservice stop > /dev/null
+
+	ret=$?
+	return $ret
+}
+
+#---------------------------------------------------------------------------------------------------
+# Helper functions.
+cleanup() {
+	cd ${CWD}
+	rm -rf ${data_dir}
+	rm -f ${schema}
+	rm -f ${tmp_file}
+}
+
+cleanup_on_finish () {
+	# change back to our dir and quit
+	cd ${CWD}
+	err=`rm -f ${schema}`
+	if [ $? -ne 0 ]; then
+		echo "${err}"
+	fi
+}
+
+set_db_trust_login() {
+	datastore_dir=$1
+	err=`cp -p ${datastore_dir}/pg_hba.conf ${datastore_dir}/pg_hba.conf.orig 2>&1`
+	if [ $? -ne 0 ]; then
+		echo "${err}"
+		return 1
+	fi
+	err=`chown ${PBS_DATA_SERVICE_USER} ${datastore_dir}/pg_hba.conf.orig`
+	if [ $? -ne 0 ]; then
+		echo "${err}"
+		return 1
+	fi
+	err=`sed 's/md5/trust/g' ${datastore_dir}/pg_hba.conf > ${datastore_dir}/pg_hba.conf.new 2>&1`
+	if [ $? -ne 0 ]; then
+		echo "${err}"
+		return 1
+	fi
+	err=`chown ${PBS_DATA_SERVICE_USER} ${datastore_dir}/pg_hba.conf.new`
+	if [ $? -ne 0 ]; then
+		echo "${err}"
+		return 1
+	fi
+	err=`mv ${datastore_dir}/pg_hba.conf.new ${datastore_dir}/pg_hba.conf 2>&1`
+	if [ $? -ne 0 ]; then
+		echo "${err}"
+		return 1
+	fi
+}
+
+revoke_db_trust_login() {
+	datastore_dir=$1
+	err=`cp -p ${datastore_dir}/pg_hba.conf.orig ${datastore_dir}/pg_hba.conf 2>&1`
+	if [ $? -eq 0 ]; then
+		rm -f ${datastore_dir}/pg_hba.conf.orig
+	else
+		echo "${err}"
+		return 1
+	fi
+}
+
+utf8_to_sql_ascii() {
```

Comment:
Which was our last utf8 database distro version - was it within the last 2 versions of PBS? If not, we should remove this code.

Reply:
I think I added it when we wanted to move from PG8.3 to 9.3 and it was in PBS v13. I guess this can go.